Restore neuvector MI

timja opened the pull request.

timja committed:
Restore neuvector MI

hmcts-platform-operations commented:
## Plan Result (ptl)




<pre><code>Plan: 0 to add, 5 to change, 0 to destroy.</code></pre>

* Update
  * azurerm_resource_group.rg
  * azurerm_storage_account.stg
  * azurerm_user_assigned_identity.managed_identity
  * module.azurekeyvault.azurerm_key_vault.kv
  * module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds




<details><summary>Change Result (Click me)</summary>

```hcl

  # azurerm_resource_group.rg will be updated in-place
  ~ resource "azurerm_resource_group" "rg" {
        id       = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/cft-neuvector-ptl-rg"
        name     = "cft-neuvector-ptl-rg"
      ~ tags     = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (1 unchanged attribute hidden)
    }

  # azurerm_storage_account.stg will be updated in-place
  ~ resource "azurerm_storage_account" "stg" {
        id                                = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/cft-neuvector-ptl-rg/providers/Microsoft.Storage/storageAccounts/cftneuvectorptl"
        name                              = "cftneuvectorptl"
      ~ tags                              = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (37 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }

  # azurerm_user_assigned_identity.managed_identity will be updated in-place
  ~ resource "azurerm_user_assigned_identity" "managed_identity" {
        id                  = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/managed-identities-cftptl-intsvc-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/neuvector-cftptl-intsvc-mi"
        name                = "neuvector-cftptl-intsvc-mi"
      ~ tags                = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (5 unchanged attributes hidden)
    }

  # module.azurekeyvault.azurerm_key_vault.kv will be updated in-place
  ~ resource "azurerm_key_vault" "kv" {
        id                              = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/cft-neuvector-ptl-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-ptl"
        name                            = "cftneuvector-ptl"
      ~ tags                            = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (13 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds will be updated in-place
  ~ resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
        id                             = "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/cft-neuvector-ptl-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-ptl|cftneuvector-ptl"
        name                           = "cftneuvector-ptl"
        # (3 unchanged attributes hidden)

      - metric {
          - category = "AllMetrics" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 5 to change, 0 to destroy.
```

</details>





<!-- github-comment: {"Command":"plan","PRNumber":20,"Program":"tfcmt","SHA1":"","Target":"ptl","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (perftest)




<pre><code>Plan: 0 to add, 5 to change, 0 to destroy.</code></pre>

* Update
  * azurerm_resource_group.rg
  * azurerm_storage_account.stg
  * azurerm_user_assigned_identity.managed_identity
  * module.azurekeyvault.azurerm_key_vault.kv
  * module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds




<details><summary>Change Result (Click me)</summary>

```hcl

  # azurerm_resource_group.rg will be updated in-place
  ~ resource "azurerm_resource_group" "rg" {
        id       = "/subscriptions/8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c/resourceGroups/cft-neuvector-perftest-rg"
        name     = "cft-neuvector-perftest-rg"
      ~ tags     = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "testing"
            "startupMode"  = "always"
        }
        # (1 unchanged attribute hidden)
    }

  # azurerm_storage_account.stg will be updated in-place
  ~ resource "azurerm_storage_account" "stg" {
        id                                = "/subscriptions/8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c/resourceGroups/cft-neuvector-perftest-rg/providers/Microsoft.Storage/storageAccounts/cftneuvectorperftest"
        name                              = "cftneuvectorperftest"
      ~ tags                              = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "testing"
            "startupMode"  = "always"
        }
        # (37 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }

  # azurerm_user_assigned_identity.managed_identity will be updated in-place
  ~ resource "azurerm_user_assigned_identity" "managed_identity" {
        id                  = "/subscriptions/7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c/resourceGroups/managed-identities-perftest-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/neuvector-perftest-mi"
        name                = "neuvector-perftest-mi"
      ~ tags                = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "testing"
            "startupMode"  = "always"
        }
        # (5 unchanged attributes hidden)
    }

  # module.azurekeyvault.azurerm_key_vault.kv will be updated in-place
  ~ resource "azurerm_key_vault" "kv" {
        id                              = "/subscriptions/8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c/resourceGroups/cft-neuvector-perftest-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-perftest"
        name                            = "cftneuvector-perftest"
      ~ tags                            = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "testing"
            "startupMode"  = "always"
        }
        # (13 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds will be updated in-place
  ~ resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
        id                             = "/subscriptions/8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c/resourceGroups/cft-neuvector-perftest-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-perftest|cftneuvector-perftest"
        name                           = "cftneuvector-perftest"
        # (3 unchanged attributes hidden)

      - metric {
          - category = "AllMetrics" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 5 to change, 0 to destroy.
```

</details>





<!-- github-comment: {"Command":"plan","PRNumber":20,"Program":"tfcmt","SHA1":"","Target":"perftest","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (aat)




<pre><code>Plan: 0 to add, 5 to change, 0 to destroy.</code></pre>

* Update
  * azurerm_resource_group.rg
  * azurerm_storage_account.stg
  * azurerm_user_assigned_identity.managed_identity
  * module.azurekeyvault.azurerm_key_vault.kv
  * module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds




<details><summary>Change Result (Click me)</summary>

```hcl

  # azurerm_resource_group.rg will be updated in-place
  ~ resource "azurerm_resource_group" "rg" {
        id       = "/subscriptions/96c274ce-846d-4e48-89a7-d528432298a7/resourceGroups/cft-neuvector-aat-rg"
        name     = "cft-neuvector-aat-rg"
      ~ tags     = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "staging"
            "startupMode"  = "always"
        }
        # (1 unchanged attribute hidden)
    }

  # azurerm_storage_account.stg will be updated in-place
  ~ resource "azurerm_storage_account" "stg" {
        id                                = "/subscriptions/96c274ce-846d-4e48-89a7-d528432298a7/resourceGroups/cft-neuvector-aat-rg/providers/Microsoft.Storage/storageAccounts/cftneuvectoraat"
        name                              = "cftneuvectoraat"
      ~ tags                              = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "staging"
            "startupMode"  = "always"
        }
        # (37 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }

  # azurerm_user_assigned_identity.managed_identity will be updated in-place
  ~ resource "azurerm_user_assigned_identity" "managed_identity" {
        id                  = "/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9/resourceGroups/managed-identities-aat-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/neuvector-aat-mi"
        name                = "neuvector-aat-mi"
      ~ tags                = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "staging"
            "startupMode"  = "always"
        }
        # (5 unchanged attributes hidden)
    }

  # module.azurekeyvault.azurerm_key_vault.kv will be updated in-place
  ~ resource "azurerm_key_vault" "kv" {
        id                              = "/subscriptions/96c274ce-846d-4e48-89a7-d528432298a7/resourceGroups/cft-neuvector-aat-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-aat"
        name                            = "cftneuvector-aat"
      ~ tags                            = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "staging"
            "startupMode"  = "always"
        }
        # (13 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds will be updated in-place
  ~ resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
        id                             = "/subscriptions/96c274ce-846d-4e48-89a7-d528432298a7/resourceGroups/cft-neuvector-aat-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-aat|cftneuvector-aat"
        name                           = "cftneuvector-aat"
        # (3 unchanged attributes hidden)

      - metric {
          - category = "AllMetrics" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 5 to change, 0 to destroy.
```

</details>





<!-- github-comment: {"Command":"plan","PRNumber":20,"Program":"tfcmt","SHA1":"","Target":"aat","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (prod)




<pre><code>Plan: 0 to add, 5 to change, 0 to destroy.</code></pre>

* Update
  * azurerm_resource_group.rg
  * azurerm_storage_account.stg
  * azurerm_user_assigned_identity.managed_identity
  * module.azurekeyvault.azurerm_key_vault.kv
  * module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds




<details><summary>Change Result (Click me)</summary>

```hcl

  # azurerm_resource_group.rg will be updated in-place
  ~ resource "azurerm_resource_group" "rg" {
        id       = "/subscriptions/8cbc6f36-7c56-4963-9d36-739db5d00b27/resourceGroups/cft-neuvector-prod-rg"
        name     = "cft-neuvector-prod-rg"
      ~ tags     = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (1 unchanged attribute hidden)
    }

  # azurerm_storage_account.stg will be updated in-place
  ~ resource "azurerm_storage_account" "stg" {
        id                                = "/subscriptions/8cbc6f36-7c56-4963-9d36-739db5d00b27/resourceGroups/cft-neuvector-prod-rg/providers/Microsoft.Storage/storageAccounts/cftneuvectorprod"
        name                              = "cftneuvectorprod"
      ~ tags                              = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (37 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }

  # azurerm_user_assigned_identity.managed_identity will be updated in-place
  ~ resource "azurerm_user_assigned_identity" "managed_identity" {
        id                  = "/subscriptions/8999dec3-0104-4a27-94ee-6588559729d1/resourceGroups/managed-identities-prod-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/neuvector-prod-mi"
        name                = "neuvector-prod-mi"
      ~ tags                = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (5 unchanged attributes hidden)
    }

  # module.azurekeyvault.azurerm_key_vault.kv will be updated in-place
  ~ resource "azurerm_key_vault" "kv" {
        id                              = "/subscriptions/8cbc6f36-7c56-4963-9d36-739db5d00b27/resourceGroups/cft-neuvector-prod-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-prod"
        name                            = "cftneuvector-prod"
      ~ tags                            = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "High"
            "environment"  = "production"
            "startupMode"  = "always"
        }
        # (13 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds will be updated in-place
  ~ resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
        id                             = "/subscriptions/8cbc6f36-7c56-4963-9d36-739db5d00b27/resourceGroups/cft-neuvector-prod-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-prod|cftneuvector-prod"
        name                           = "cftneuvector-prod"
        # (3 unchanged attributes hidden)

      - metric {
          - category = "AllMetrics" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 5 to change, 0 to destroy.
```

</details>





<!-- github-comment: {"Command":"plan","PRNumber":20,"Program":"tfcmt","SHA1":"","Target":"prod","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (ithc)




<pre><code>Plan: 0 to add, 5 to change, 0 to destroy.</code></pre>

* Update
  * azurerm_resource_group.rg
  * azurerm_storage_account.stg
  * azurerm_user_assigned_identity.managed_identity
  * module.azurekeyvault.azurerm_key_vault.kv
  * module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds




<details><summary>Change Result (Click me)</summary>

```hcl

  # azurerm_resource_group.rg will be updated in-place
  ~ resource "azurerm_resource_group" "rg" {
        id       = "/subscriptions/62864d44-5da9-4ae9-89e7-0cf33942fa09/resourceGroups/cft-neuvector-ithc-rg"
        name     = "cft-neuvector-ithc-rg"
      ~ tags     = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "ithc"
            "startupMode"  = "always"
        }
        # (1 unchanged attribute hidden)
    }

  # azurerm_storage_account.stg will be updated in-place
  ~ resource "azurerm_storage_account" "stg" {
        id                                = "/subscriptions/62864d44-5da9-4ae9-89e7-0cf33942fa09/resourceGroups/cft-neuvector-ithc-rg/providers/Microsoft.Storage/storageAccounts/cftneuvectorithc"
        name                              = "cftneuvectorithc"
      ~ tags                              = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "ithc"
            "startupMode"  = "always"
        }
        # (37 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }

  # azurerm_user_assigned_identity.managed_identity will be updated in-place
  ~ resource "azurerm_user_assigned_identity" "managed_identity" {
        id                  = "/subscriptions/7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c/resourceGroups/managed-identities-ithc-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/neuvector-ithc-mi"
        name                = "neuvector-ithc-mi"
      ~ tags                = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "ithc"
            "startupMode"  = "always"
        }
        # (5 unchanged attributes hidden)
    }

  # module.azurekeyvault.azurerm_key_vault.kv will be updated in-place
  ~ resource "azurerm_key_vault" "kv" {
        id                              = "/subscriptions/62864d44-5da9-4ae9-89e7-0cf33942fa09/resourceGroups/cft-neuvector-ithc-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-ithc"
        name                            = "cftneuvector-ithc"
      ~ tags                            = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
          ~ "businessArea" = "Cross-Cutting" -> "CFT"
            "criticality"  = "Medium"
            "environment"  = "ithc"
            "startupMode"  = "always"
        }
        # (13 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds will be updated in-place
  ~ resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
        id                             = "/subscriptions/62864d44-5da9-4ae9-89e7-0cf33942fa09/resourceGroups/cft-neuvector-ithc-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-ithc|cftneuvector-ithc"
        name                           = "cftneuvector-ithc"
        # (3 unchanged attributes hidden)

      - metric {
          - category = "AllMetrics" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 5 to change, 0 to destroy.
```

</details>





<!-- github-comment: {"Command":"plan","PRNumber":20,"Program":"tfcmt","SHA1":"","Target":"ithc","Vars":{}} -->

timja committed:
Refactor resource

Error was:

│ Error: Invalid for_each argument
│
│   on .terraform/modules/azurekeyvault/managed-identity.tf line 37, in resource "azurerm_key_vault_access_policy" "managed_identity_access_policy":
│   37:   for_each = local.managed_identity_list
│     ├────────────────
│     │ local.managed_identity_list is a set of dynamic, known only after apply
│
│ The "for_each" set includes values derived from resource attributes that
│ cannot be determined until apply, and so Terraform cannot determine the
│ full set of keys that will identify the instances of this resource.
│
│ When working with unknown values in for_each, it's better to use a map
│ value where the keys are defined statically in your configuration and where
│ only the values contain apply-time results.
│
│ Alternatively, you could use the -target planning option to first apply
│ only the resources that the for_each value depends on, and then apply a
│ second time to fully converge.

timja committed:
Remove unused resource

timja committed:
Fix tag

timja committed:
Rerun CI

hmcts-platform-operations commented:
## Plan Result (sbox)




### :warning: Resource Deletion will happen :warning:
This plan contains resource delete operation. Please check the plan result very carefully!

<pre><code>Plan: 6 to add, 2 to change, 3 to destroy.</code></pre>

* Create
  * azurerm_storage_account.stg
  * azurerm_storage_share.share[0]
  * azurerm_storage_share.share[1]
  * module.azurekeyvault.azurerm_key_vault_access_policy.managed_identity_access_policy[&#34;01e164a6-001f-406f-9708-07e540cbe0e7&#34;]
* Update
  * azurerm_key_vault_secret.kvs
  * module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds
* Delete
  * module.azurekeyvault.azurerm_key_vault_access_policy.managed_identity_access_policy[&#34;1c1c7ac9-fab4-4b11-ab9a-29e14336f15d&#34;]
* Replace
  * azurerm_key_vault_access_policy.cftapps_kv_access_policy
  * azurerm_key_vault_access_policy.managed_identity_access_policy




<details><summary>Change Result (Click me)</summary>

```hcl

  # azurerm_key_vault_access_policy.cftapps_kv_access_policy must be replaced
-/+ resource "azurerm_key_vault_access_policy" "cftapps_kv_access_policy" {
      ~ id                      = "/subscriptions/b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb/resourceGroups/core-infra-sbox-rg/providers/Microsoft.KeyVault/vaults/cftapps-sbox/objectId/1c1c7ac9-fab4-4b11-ab9a-29e14336f15d" -> (known after apply)
      ~ object_id               = "1c1c7ac9-fab4-4b11-ab9a-29e14336f15d" -> "01e164a6-001f-406f-9708-07e540cbe0e7" # forces replacement
      - storage_permissions     = [] -> null
        # (5 unchanged attributes hidden)
    }

  # azurerm_key_vault_access_policy.managed_identity_access_policy must be replaced
-/+ resource "azurerm_key_vault_access_policy" "managed_identity_access_policy" {
      ~ id                      = "/subscriptions/b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb/resourceGroups/cft-neuvector-sbox-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-sbox/objectId/1c1c7ac9-fab4-4b11-ab9a-29e14336f15d" -> (known after apply)
      ~ object_id               = "1c1c7ac9-fab4-4b11-ab9a-29e14336f15d" -> "01e164a6-001f-406f-9708-07e540cbe0e7" # forces replacement
      - storage_permissions     = [] -> null
        # (5 unchanged attributes hidden)
    }

  # azurerm_key_vault_secret.kvs will be updated in-place
  ~ resource "azurerm_key_vault_secret" "kvs" {
        id                      = "https://cftneuvector-sbox.vault.azure.net/secrets/storage-account-key/edd03e2e7eee4afeb62603fceb8beaa4"
        name                    = "storage-account-key"
        tags                    = {}
      ~ value                   = (sensitive value)
        # (5 unchanged attributes hidden)
    }

  # azurerm_storage_account.stg will be created
  + resource "azurerm_storage_account" "stg" {
      + access_tier                       = (known after apply)
      + account_kind                      = "StorageV2"
      + account_replication_type          = "LRS"
      + account_tier                      = "Standard"
      + allow_nested_items_to_be_public   = true
      + cross_tenant_replication_enabled  = true
      + default_to_oauth_authentication   = false
      + enable_https_traffic_only         = true
      + id                                = (known after apply)
      + infrastructure_encryption_enabled = false
      + is_hns_enabled                    = false
      + large_file_share_enabled          = (known after apply)
      + location                          = "uksouth"
      + min_tls_version                   = "TLS1_2"
      + name                              = "cftneuvectorsbox"
      + nfsv3_enabled                     = false
      + primary_access_key                = (sensitive value)
      + primary_blob_connection_string    = (sensitive value)
      + primary_blob_endpoint             = (known after apply)
      + primary_blob_host                 = (known after apply)
      + primary_connection_string         = (sensitive value)
      + primary_dfs_endpoint              = (known after apply)
      + primary_dfs_host                  = (known after apply)
      + primary_file_endpoint             = (known after apply)
      + primary_file_host                 = (known after apply)
      + primary_location                  = (known after apply)
      + primary_queue_endpoint            = (known after apply)
      + primary_queue_host                = (known after apply)
      + primary_table_endpoint            = (known after apply)
      + primary_table_host                = (known after apply)
      + primary_web_endpoint              = (known after apply)
      + primary_web_host                  = (known after apply)
      + public_network_access_enabled     = true
      + queue_encryption_key_type         = "Service"
      + resource_group_name               = "cft-neuvector-sbox-rg"
      + secondary_access_key              = (sensitive value)
      + secondary_blob_connection_string  = (sensitive value)
      + secondary_blob_endpoint           = (known after apply)
      + secondary_blob_host               = (known after apply)
      + secondary_connection_string       = (sensitive value)
      + secondary_dfs_endpoint            = (known after apply)
      + secondary_dfs_host                = (known after apply)
      + secondary_file_endpoint           = (known after apply)
      + secondary_file_host               = (known after apply)
      + secondary_location                = (known after apply)
      + secondary_queue_endpoint          = (known after apply)
      + secondary_queue_host              = (known after apply)
      + secondary_table_endpoint          = (known after apply)
      + secondary_table_host              = (known after apply)
      + secondary_web_endpoint            = (known after apply)
      + secondary_web_host                = (known after apply)
      + sftp_enabled                      = false
      + shared_access_key_enabled         = true
      + table_encryption_key_type         = "Service"
      + tags                              = {
          + "application"  = "core"
          + "builtFrom"    = "hmcts/cft-neuvector"
          + "businessArea" = "CFT"
          + "criticality"  = "Low"
          + "environment"  = "sandbox"
          + "expiresAfter" = "3000-01-01"
          + "startupMode"  = "always"
        }
    }

  # azurerm_storage_share.share[0] will be created
  + resource "azurerm_storage_share" "share" {
      + access_tier          = (known after apply)
      + enabled_protocol     = "SMB"
      + id                   = (known after apply)
      + metadata             = (known after apply)
      + name                 = "neuvector-data-00"
      + quota                = 1
      + resource_manager_id  = (known after apply)
      + storage_account_name = "cftneuvectorsbox"
      + url                  = (known after apply)
    }

  # azurerm_storage_share.share[1] will be created
  + resource "azurerm_storage_share" "share" {
      + access_tier          = (known after apply)
      + enabled_protocol     = "SMB"
      + id                   = (known after apply)
      + metadata             = (known after apply)
      + name                 = "neuvector-data-01"
      + quota                = 1
      + resource_manager_id  = (known after apply)
      + storage_account_name = "cftneuvectorsbox"
      + url                  = (known after apply)
    }

  # module.azurekeyvault.azurerm_key_vault_access_policy.managed_identity_access_policy["01e164a6-001f-406f-9708-07e540cbe0e7"] will be created
  + resource "azurerm_key_vault_access_policy" "managed_identity_access_policy" {
      + certificate_permissions = [
          + "Get",
          + "List",
        ]
      + id                      = (known after apply)
      + key_permissions         = [
          + "Get",
          + "List",
        ]
      + key_vault_id            = "/subscriptions/b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb/resourceGroups/cft-neuvector-sbox-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-sbox"
      + object_id               = "01e164a6-001f-406f-9708-07e540cbe0e7"
      + secret_permissions      = [
          + "Get",
          + "List",
        ]
      + tenant_id               = "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
    }

  # module.azurekeyvault.azurerm_key_vault_access_policy.managed_identity_access_policy["1c1c7ac9-fab4-4b11-ab9a-29e14336f15d"] will be destroyed
  # (because key ["1c1c7ac9-fab4-4b11-ab9a-29e14336f15d"] is not in for_each map)
  - resource "azurerm_key_vault_access_policy" "managed_identity_access_policy" {
      - certificate_permissions = [
          - "Get",
          - "List",
        ] -> null
      - id                      = "/subscriptions/b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb/resourceGroups/cft-neuvector-sbox-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-sbox/objectId/1c1c7ac9-fab4-4b11-ab9a-29e14336f15d" -> null
      - key_permissions         = [
          - "Get",
          - "List",
        ] -> null
      - key_vault_id            = "/subscriptions/b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb/resourceGroups/cft-neuvector-sbox-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-sbox" -> null
      - object_id               = "1c1c7ac9-fab4-4b11-ab9a-29e14336f15d" -> null
      - secret_permissions      = [
          - "Get",
          - "List",
        ] -> null
      - storage_permissions     = [] -> null
      - tenant_id               = "531ff96d-0ae9-462a-8d2d-bec7c0b42082" -> null
    }

  # module.azurekeyvault.azurerm_monitor_diagnostic_setting.kv-ds will be updated in-place
  ~ resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
        id                             = "/subscriptions/b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb/resourceGroups/cft-neuvector-sbox-rg/providers/Microsoft.KeyVault/vaults/cftneuvector-sbox|cftneuvector-sbox"
        name                           = "cftneuvector-sbox"
        # (3 unchanged attributes hidden)

      - metric {
          - category = "AllMetrics" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

        # (3 unchanged blocks hidden)
    }

Plan: 6 to add, 2 to change, 3 to destroy.
```

</details>



<details><summary>:information_source: Objects have changed outside of Terraform</summary>

_This feature was introduced from [Terraform v0.15.4](https://github.com/hashicorp/terraform/releases/tag/v0.15.4)._

```hcl

Terraform detected the following changes made outside of Terraform since the
last "terraform apply" which may have affected this plan:

  # azurerm_storage_account.stg has been deleted
  - resource "azurerm_storage_account" "stg" {
        id                                = "/subscriptions/b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb/resourceGroups/cft-neuvector-sbox-rg/providers/Microsoft.Storage/storageAccounts/cftneuvectorsbox"
      - name                              = "cftneuvectorsbox" -> null
      - primary_access_key                = (sensitive value) -> null
        tags                              = {
            "application"  = "core"
            "builtFrom"    = "hmcts/cft-neuvector"
            "businessArea" = "Cross-Cutting"
            "criticality"  = "Low"
            "environment"  = "sandbox"
            "expiresAfter" = "2023-12-08"
            "startupMode"  = "always"
        }
        # (36 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }


Unless you have made equivalent changes to your configuration, or ignored the
```

</details>



<!-- github-comment: {"Command":"plan","PRNumber":20,"Program":"tfcmt","SHA1":"","Target":"sbox","Vars":{}} -->
